aiofcm/connection.py

```diff
@@ -147,6 +162,26 @@ async def send_message(self, message):
         response = await future_response
         return response
 
+    async def send_ack(self, device_token, message_id):
+        if not self.connected:
+            await self.connect()
+        msg = aioxmpp.Message(
+            type_=aioxmpp.MessageType.NORMAL
+        )
+        payload = FCMMessage()
+
+        payload_body = {"to":str(device_token), "message_id":str(message_id), "message_type":"ack"}
+
+        payload.text = json.dumps(payload_body)
+        msg.fcm_payload = payload
+
+        self.refresh_inactivity_timer()
+        try:
+            await self.xmpp_client.stream.send(msg)
+        except Exception:
+            self.requests.pop(message.message_id)
```

Comment:
`NameError` will be raised but anyway why do we need `pop` here? `self.requests` contains futures for downstream messages.

Reply:
Agree. The pop must be removed. (Therefore the try/except control-flow can also be removed)
Fixed.